site_scons/site_init.py

```diff
@@ -0,0 +1,25 @@
+"""Configure/setup global scons options for DAOS"""
+
+import shutil
+import sys
+import os
+
+def set_auto_clean():
+    """Replicate using --config=force on the command line"""
+
+    if (GetOption('config') == 'auto' and
+        not GetOption('clean') and
+        not GetOption('help')):
+
+        # If a config value is set then do not wipe the config cache here.
+        for arg in sys.argv:
+            if arg.startswith('--config='):
+                return
+
+        if os.path.exists('.sconsign.dblite'):
```

Comment:
as discussed, I think it's better to leave this out

Reply:
Yes, I agree.  It's a often requested feature however and I do know how to do it now, both in the code and in peoples . files.  I propose we remove the site_init.py from this PR but keep the rest of it (mainly WhereIs and pylint changes) and document how to make config=force the default if required.

I still maintain that if --config=force is required in normal workflow however that it's probably a bug with how dependencies are setup, wether that's with our code or with scons itself.